src/MVDegrains.cpp

```diff
@@ -96,96 +96,22 @@ static const VSFrameRef *VS_CC mvdegrainGetFrame(int n, int activationReason, vo
-        }
-
-        if (radius > 4) {
-            int offF5 = -1 * d->vectors_data[Forward5].nDeltaFrame;
-            if (n + offF5 >= 0)
-                vsapi->requestFrameFilter(n + offF5, d->super, frameCtx);
-        }
-
-        if (radius > 3) {
-            int offF4 = -1 * d->vectors_data[Forward4].nDeltaFrame;
-            if (n + offF4 >= 0)
-                vsapi->requestFrameFilter(n + offF4, d->super, frameCtx);
-        }
-
-        if (radius > 2) {
-            int offF3 = -1 * d->vectors_data[Forward3].nDeltaFrame;
-            if (n + offF3 >= 0)
-                vsapi->requestFrameFilter(n + offF3, d->super, frameCtx);
-        }
-
-        if (radius > 1) {
-            int offF2 = -1 * d->vectors_data[Forward2].nDeltaFrame;
-            if (n + offF2 >= 0)
-                vsapi->requestFrameFilter(n + offF2, d->super, frameCtx);
-        }
-
-        int offF = -1 * d->vectors_data[Forward1].nDeltaFrame;
-        if (n + offF >= 0)
-            vsapi->requestFrameFilter(n + offF, d->super, frameCtx);
-
-        int offB = d->vectors_data[Backward1].nDeltaFrame;
-        if (n + offB < d->vi->numFrames)
-            vsapi->requestFrameFilter(n + offB, d->super, frameCtx);
-
-        if (radius > 1) {
-            int offB2 = d->vectors_data[Backward2].nDeltaFrame;
-            if (n + offB2 < d->vi->numFrames)
-                vsapi->requestFrameFilter(n + offB2, d->super, frameCtx);
-        }
-
-        if (radius > 2) {
-            int offB3 = d->vectors_data[Backward3].nDeltaFrame;
-            if (n + offB3 < d->vi->numFrames)
-                vsapi->requestFrameFilter(n + offB3, d->super, frameCtx);
-        }
-
-        if (radius > 3) {
-            int offB4 = d->vectors_data[Backward4].nDeltaFrame;
-            if (n + offB4 < d->vi->numFrames)
-                vsapi->requestFrameFilter(n + offB4, d->super, frameCtx);
-        }
-
-        if (radius > 4) {
-            int offB5 = d->vectors_data[Backward5].nDeltaFrame;
-            if (n + offB5 < d->vi->numFrames)
-                vsapi->requestFrameFilter(n + offB5, d->super, frameCtx);
-        }
 
-        if (radius > 5) {
-            int offB6 = d->vectors_data[Backward6].nDeltaFrame;
-            if (n + offB6 < d->vi->numFrames)
-                vsapi->requestFrameFilter(n + offB6, d->super, frameCtx);
+        for (int r = 0; r < radius * 2; r += 2) {
+            //Backward
+            vsapi->requestFrameFilter(n, d->vectors[r], frameCtx);
+            //Forward
+            vsapi->requestFrameFilter(n, d->vectors[r + 1], frameCtx);
+
+            // Backward
+            int offB = d->vectors_data[r].nDeltaFrame;
+            if (n + offB < d->vi->numFrames)
+                vsapi->requestFrameFilter(n + offB, d->super, frameCtx);
+
+            // Forward
+            int offF = -1 * d->vectors_data[r + 1].nDeltaFrame;
+            if (n + offF >= 0)
+                vsapi->requestFrameFilter(n + offF, d->super, frameCtx);
```

Comment:
In this case it seems to be fine somehow, but I think frames should always be requested in ascending order for best performance. Anyway, if it works it's fine.

Reply:
It does work just fine, but it shouldn’t be too much work to get it making requests in ascending order, so I’ll take a look at that.

Reply:
It occurs to me - you said ascending order. I’m presuming you mean furthest frame backwards up to furthest frame forwards? So -radius to +radius

That kinda makes sense to me, but it looks like the old code started with the furthest frame forward and then worked its way backwards. So +radius to -radius.

Reply:
Ascending order: 0, 1, 2, 3...

Backward vectors are calculated from ``n + delta`` *backwards* to ``n``.

Forward vectors are calculated from ``n - delta`` *forwards* to ``n``.

Reply:
Ahhh thanks for the correction there - I was inverting these in my head, thinking that they corresponded to previous/next respectively, but it's in fact inverted.

I'll update this PR with proper traversal from forward to backward instead.

Reply:
I tried this combo six way from sunday, and could find no discernible difference in performance between the old code, this code, and requesting forwards to n to backwards.

This was with 4k, 1080p, and 540p clips with 8, and 16 blksize.

I have the benchmarks to prove it, if desired.

I just left this code as is, since its the simplest implementation and works just fine.